### ISSUE TITLE
Couple of performance tweaks

### ISSUE THREAD

**Issue by tipara** [matterport/Mask_RCNN#416]
Hello,

First of all, I'd like to thank the devs for their great work on this project. Thought I'd contribute a tiny bit by suggesting two very simple performance tweaks:

1/ The current model.train function in model.py (purposedly) uses only 1 worker thread on Windows (not tested on Linux) due to a "bug" in Keras with use_multiprocessing=True. However, it is possible to call Keras's fit_generator with multiple workers and use_multiprocessing = False. In my use case (training on 30k full HD images, Win 10, Keras 2.1.5), using:
`workers = multiprocessing.cpu_count()
`
resulted in a ~30% speedup on each epoch (single 1080 Ti with 6700k CPU). I suspect performance can still be improved, as average GPU usage is quite low, but it's already a nice step-up.

2/ In visualize.py, the apply_mask function performs a where on the same image-sized mask three times. No big deal when calling it once or twice, but not great when called in a loop over 3k validation images. Instead of the current code, I suggest:
```
def apply_mask(image, mask, color, alpha=0.5):
    """Apply the given mask to the image.
    """
    mask_px = np.where(mask)
    for c in range(3):
        image[mask_px[0], mask_px[1], c] = (1 - alpha)*image[mask_px[0], mask_px[1], c] + alpha * color[c] * 255
    return image
```

Hoping this will be helpful to someone

**Comment by GustavZ** [comment]
@tipara could you tell in detail what you changed for tweak 1?

As i understood, you set use_multiprocessing to False and thats it? because workers is already set to multiprocessing.cpu_count()

**Comment by tipara** [comment]
Yes, for tweak 1 you just have to set `use_multiprocessing = False` and remove the condition on os.name. Basically, you just have to replace [lines 2313 to 2329](https://github.com/matterport/Mask_RCNN/blob/4129a27275c48c672f6fd8c6303a88ba1eed643b/mrcnn/model.py#L2313) with:

```
        workers = multiprocessing.cpu_count()

        self.keras_model.fit_generator(
            train_generator,
            initial_epoch=self.epoch,
            epochs=epochs,
            steps_per_epoch=self.config.STEPS_PER_EPOCH,
            callbacks=callbacks,
            validation_data=val_generator,
            validation_steps=self.config.VALIDATION_STEPS,
            max_queue_size=100,
            workers=workers,
            use_multiprocessing=False,
        )
```
